src/Databases/DatabaseReplicated.cpp

```diff
@@ -1299,7 +1300,7 @@ bool DatabaseReplicated::canExecuteReplicatedMetadataAlter() const
     /// It may update the metadata digest (both locally and in ZooKeeper)
     /// before DatabaseReplicatedDDLWorker::initializeReplication() has finished.
     /// We should not update metadata until the database is initialized.
-    return ddl_worker && ddl_worker->isCurrentlyActive();
+    return ddl_worker_initialized && ddl_worker->isCurrentlyActive();
```

Comment:
It may crash on shutdown:
https://github.com/ClickHouse/ClickHouse/blob/8013cb1f784f6324b3c7b227499751dc7e666009/src/Databases/DatabaseReplicated.cpp#L1158-L1159